syntax = "proto2";

package yolov3;

import "protos/convolution.proto";
import "protos/residual.proto";
import "protos/route.proto";
import "protos/upsample.proto";
import "protos/yolo.proto";
import "protos/eval.proto";
import "protos/train.proto";
import "protos/modelspecs.proto";
import "protos/classmap.proto";

message Network {
    repeated Convolution convolutions = 1;
    repeated ResidualBlock residuals = 2;
    repeated Route routes = 3;
    repeated UpSample upsampling = 4;
    repeated Yolo yolos = 5;
};

message NetworkDefinition {
    required Network network_def = 1;
    required Train train_config = 2;
    required Eval eval_config = 3;
    required ModelSpecs model_specs = 4;
    required ClassMap class_def = 5;
}
